value/matrix.go

```diff
@@ -495,3 +495,52 @@ func (m *Matrix) dtranspose(c Context, v Vector) *Matrix {
 
 	return NewMatrix(shape, data)
 }
+
+// sel returns the selection of m according to v.
+// Like in APL, the selection applies to the final axis.
+func (m *Matrix) sel(c Context, v Vector) *Matrix {
+	// All lhs values must be small integers.
+	var count int64
+	for _, x := range v {
```

Comment:
vector.go has AllChars. maybe we need AllInts, or perhaps even a way to have a general one, although it's not easy without reflection, which i don't want. but AllInts would be consistent.

Reply:
Done as AllInts. The general one would be

```
func VectorAll[T any](v Vector) bool {
    for _, c := range v {
        if _, ok := c.Inner().(T); !ok {
            return false
        }
    }
    return true
}

if !VectorAll[Int](v) { ... }
```

(It can't be a method on Vector because there are no parameterized methods, because then interface satisfaction is either inconsistent or incomplete.)